src/main/java/seedu/taassist/model/moduleclass/ModuleClass.java

```diff
@@ -76,22 +75,17 @@ public List<Session> getSessions() {
      */
     public ModuleClass addSession(Session session) {
         requireNonNull(session);
-        if (hasSession(session)) {
-            return this;
-        }
-        List<Session> newSessions = new ArrayList<>(sessions);
-        newSessions.add(session);
-        return new ModuleClass(className, newSessions);
+        sessions.add(session);
+        return this;
```

Comment:
`ModuleClass`-es must be immutable, this guarantee must not change so that updates to UI still propagate correctly.

Reply:
This is reverted in the later commit.